src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -113,6 +114,15 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     SnapshotRequestModel snapshotReq =
         inputParameters.get(JobMapKeys.REQUEST.getKeyName(), SnapshotRequestModel.class);
     String snapshotName = snapshotReq.getName();
+    SnapshotRequestContentsModel contents = snapshotReq.getContents().get(0);
+    SnapshotRequestContentsModel.ModeEnum mode = contents.getMode();
+
+    // at start of flight, store the flight id in the snapshot request
+    if (mode == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
+      addStep(
+          new AddFlightIdToSnapshotRequestStep(
+              snapshotRequestDao, contents.getRequestIdSpec().getSnapshotRequestId()));
```

Comment:
I would also create a local variable for `contents.getRequestIdSpec().getSnapshotRequestId()`
```suggestion
              snapshotRequestDao, snapshotRequestId));
```

Although I see that `CreateSnapshotByRequestIdInterface` also needs the request ID but it extracts the request ID from `snapshotReq`. Should these new steps also work this way?